src/macros/@production.jl

```diff
@@ -15,6 +15,15 @@ function prune!(T::AbstractArray{<:Any})
     return prune!.(T)
 end
 
+function prune!(T::abstractDemandFlow)
+    if quantity(T) != 0 && base_quantity(T) != 0
+        return T
+    else
+        return nothing
+    end
+end
+
+
```

Comment:
We should also prune demands. Get rid of demands/endowments with zero base quantity.